core/codenvy-hosted-workspace-activity/src/test/java/com/codenvy/activity/server/WorkspaceActivityServiceTest.java

```diff
@@ -0,0 +1,135 @@
+import org.eclipse.che.api.core.model.workspace.WorkspaceStatus;
+import org.eclipse.che.api.core.rest.ApiExceptionMapper;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceConfigImpl;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.eclipse.che.commons.env.EnvironmentContext;
+import org.eclipse.che.commons.user.User;
+import org.eclipse.che.commons.user.UserImpl;
+import org.everrest.assured.EverrestJetty;
+import org.everrest.core.Filter;
+import org.everrest.core.GenericContainerRequest;
+import org.everrest.core.RequestFilter;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * @author Mihail Kuznyetsov
+ */
+@Listeners(value = {EverrestJetty.class, MockitoTestNGListener.class})
+public class WorkspaceActivityServiceTest {
+
+    @SuppressWarnings("unused") // used by EverrestJetty
+    private static final EnvironmentFilter  FILTER           = new EnvironmentFilter();
+    @SuppressWarnings("unused") // used by EverrestJetty
+    private final ApiExceptionMapper exceptionMapper    = new ApiExceptionMapper();
+
+    private static final String SERVICE_PATH = "/activity";
+    private static final String USER_ID      = "user123";
+    private static final String WORKSPACE_ID = "workspace123";
+    private static final User   TEST_USER        = new UserImpl("name", USER_ID, "token", null, false);
+    @Mock
+    private WorkspaceActivityManager workspaceActivityManager;
+
+    @Mock
+    private WorkspaceManager workspaceManager;
+
+    private WorkspaceActivityService workspaceActivityService;
+
+    @BeforeMethod
+    public void setUp() {
+        workspaceActivityService = new WorkspaceActivityService(workspaceActivityManager, workspaceManager);
+    }
+
+    @Test
+    public void shouldUpdateWorkspaceActivityOnRunningWorkspace() throws NotFoundException, ServerException {
+        // given
+        final WorkspaceImpl workspace = createWorkspace(USER_ID, WorkspaceStatus.RUNNING);
+        when(workspaceManager.getWorkspace(WORKSPACE_ID)).thenReturn(workspace);
+
+        // when
+        Response response = given().when().put(SERVICE_PATH + '/' + WORKSPACE_ID);
+
+        // then
+        assertEquals(response.getStatusCode(), 204);
+        verify(workspaceActivityManager).update(eq(WORKSPACE_ID), anyLong());
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenUpdatingWorkspaceActivity() throws NotFoundException, ServerException {
```

Comment:
When I see name of this method method I think that it check whether tested class throws exception on any activity or not.
Please rename it to show the real usecase when exception is thrown